### ISSUE TITLE
BOTH THEMES: Favicon location

### ISSUE THREAD

**Issue by KathJohnson** [meetinghouse/cms#223]
Can you tell me where the favicon is saved for all the websites?

I'm not asking to make it accessible through the interface, but I would like to know where it is stored so we can update people's favicons as needed. Thanks.

**Comment by jboz62** [comment]
Hmm, good question. According to the source code, it's supposed to be in /docs-assets/ico/favicon.png, but I don't even see those directories, do you? You might have to create them if they don't.

**Comment by jboz62** [comment]
You could just change the location to wherever you want, the link is in main.blade.php:

<!DOCTYPE html>
<html lang="en" ng-app="app">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="shortcut icon" href="../../docs-assets/ico/favicon.png">

Usually, when you have a ../ that means you're going back 2 directories, doesn't seem to be doing that here, though. Could be that this is deeper in the directories somewhere. Can we search Github?

**Comment by KathJohnson** [comment]
Thanks. I'll close this one now.

**Comment by jboz62** [comment]
Ok, we can search on here, nice. So it looks like the actual path is /public/assets/ico/favicon.png, but the path that is indicated in the main.blade.php seems to be wrong: /docs-assets/ico/favicon.png

**Comment by KathJohnson** [comment]
Oh, interesting. I'll upload a file to the actual location and see what happens. If nothing shows up, I'll try changing the path in the main-blade and redeploy code. I'll let you know.

**Comment by meetinghouse** [comment]
So, I uploaded a file to the actual location and nothing showed up.

Then I changed the path in the main.blade.php file and pushed to the testdeploy site and got an error.

So I changed the code back, but unfortunately, I am still getting the error, so now I feel like an idiot for rushing ahead with this.

![screenshot 2016-02-23 at 2 23 09 pm](https://cloud.githubusercontent.com/assets/5215893/13263759/fef25fd2-da38-11e5-91fc-acfc7d555fb6.png)

**Comment by jboz62** [comment]
Not to worry, looks like you have an extra '<' in your code. See if you can find that and remove it.

**Comment by jboz62** [comment]
It looks like there was some kind of mismatch in the page controller file, at least I think that what this means:

<<<<<<< HEAD
		return Redirect::route('pages.index');
=======
        return Redirect::route('pages.index');
>>>>>>> a51f216970df317c8b8e7b8f99f0cac9dcde3de1

**Comment by jboz62** [comment]
Did you make your changes via git? Wonder if it is saying there's a mismatch in the files somewhere. Might need to ask Andy on this one. I've run into this using Subversion, but I'm new to Github.

Did you figure anything out?

**Comment by meetinghouse** [comment]
Yes, I made my chage via git, but I think I can see what happened.

You know, when I looked in Forge, I saw that testdeploy had been changed to deploy the development branch. That makes sense, since y'all were using it for that specific project. I changed it in forge to deploy from the blog branch, and now all is right again with testdeploy. Whew!

But I still need to try changing the filepath again. I don't understand the /../.. part of the path, though, so I am a little nervous at this point to make that change.

Would I merely change the last line (line 9) to
<link rel="shortcut icon" href="../../public/assets/ico/favicon.png">

**Comment by jboz62** [comment]
Hmm, made a comment on this in the code, but I don't see it now. I think you just need to experiment with what is going to work with the path. The index.php page is in the 'public' directory, so we'd start with 'assests', but I'm not sure how this system handles the `../../`. So, I would try either `<link rel="shortcut icon" href="../../assets/ico/favicon.png">` or `<link rel="shortcut icon" href="../assets/ico/favicon.png">` or lastly, `<link rel="shortcut icon" href="/assets/ico/favicon.png">`. If you do the first and it doesn't work, you can look at the source code and open the link to see where it's going, then adjust your `../../`accordingly.

**Comment by jboz62** [comment]
Sometimes you have to reload 2 or 3 times to get the favicon to show up, and if one exists it can be a bugger trying to get it to change to the new one!

**Comment by KathJohnson** [comment]
I think this is resolved now. I am having a problem with deployment, for which I've created a separate issue, so I moved my efforts to another test site I am able to deploy to: ligththemetest

I renamed the file favicon2.png and then updated the code in Git to change the end of line 9 in the main.blade.php file to
"../../assets/ico/favicon2.png"

So that's an altered version of your first suggestion. When I deploy code, it updates the favicon. So yay, this is resolved. I've added this step, with details, to the launch checklist so this should be able to go into production now. Thanks for your help.
